test/tests.py

```diff
@@ -972,6 +1017,39 @@ def test_create_file_name_conflict_dir(self):
 
         os.rmdir(testFilePath)
 
+    def test_symlink_operations(self):
+        testSymlinkDir = os.path.join(self.blobstage, "test-symlink-directory")
+        testSymlinkFile = os.path.join(self.blobstage, "test-symlink-file")
+        testDir = os.path.join(self.blobstage, "test-dir")
+
+        self.assertFalse(os.path.exists(testSymlinkDir))
+        self.assertFalse(os.path.islink(testSymlinkDir))
+
+        os.makedirs(testDir)
+        os.symlink(testDir, testSymlinkDir)
+        self.assertTrue(os.path.exists(testDir))
+        self.assertTrue(os.path.islink(testSymlinkDir))
+
+        filetxt = "Some file text here."
+        testFilePath = os.path.join(testDir, "file1")
+        with open(testFilePath, 'w') as fileblob:
+            fileblob.write(filetxt)
+
+        os.symlink(testFilePath, testSymlinkFile) 
+
+        # test accessing data with the symlink directory path
+        testFilewSymlinkPath = os.path.join(testSymlinkDir, "file1")
+        self.assertTrue(os.path.exists(testFilewSymlinkPath))
+        with open(testFilewSymlinkPath, 'r') as testFile:
+            contents = testFile.read()
+            self.assertEqual(filetxt, contents)
+
+        # test accessing the same data with the symlink file path
```

Comment:
maybe we should have a separate test directories, then file, then both combined?

Reply:
For now i have just used the existing test suite. Also there are 2 new GO scripts added to the test suite feature_test.go has some more symlink related tests and there this has been covered.